frontend/src/main/components/Quarters/SingleQuarterDropdown.js

```diff
@@ -21,6 +21,9 @@ function SingleQuarterDropdown({
   label = "Quarter",
 }) {
   const localSearchQuarter = localStorage.getItem(controlId);
+  if (!localSearchQuarter) {
+    localStorage.setItem(controlId, quarters[0].yyyyq);
```

Comment:
Components must not have side effects during render. For the reasoning see https://react.dev/learn/keeping-components-pure#side-effects-unintended-consequences.

For alternative approaches see https://github.com/ucsb-cs156-f23/proj-courses-f23-7pm-4/pull/26#discussion_r1401336976

Reply:
So what if I only use the useEffect to fix it?
function MyComponent({ controlId, quarters }) {
  useEffect(() => {
    const localSearchQuarter = localStorage.getItem(controlId);
    if (!localSearchQuarter) {
      localStorage.setItem(controlId, quarters[0].yyyyq);
    }
  }, [controlId, quarters]);

Reply:
That would work and it'd certainly be better than causing side effects during render. It would just be
```js
useEffect(() => {
  if (!localSearchQuarter) {
    localStorage.setItem(controlId, quarters[0].yyyyq);
  }
}, [controlId, quarters, localSearchQuarter]);
```
But the first alternative approach I linked is easiest and one line long and doesn't require any hook. It looks like PRs from other groups that did the same thing have been merged anyway, so at this point I'm not too picky on which approach you take.